lib/storage/src/content_manager/snapshots/mod.rs

```diff
@@ -37,108 +38,165 @@ pub async fn get_full_snapshot_path(
-    }
 
-    let full_snapshot_path = snapshot_dir.join(&snapshot_name);
-
-    let config_path_clone = config_path.clone();
-    let full_snapshot_path_clone = full_snapshot_path.clone();
-    let created_snapshots_clone: Vec<_> = created_snapshots
-        .iter()
-        .map(|(x, y)| (x.to_string(), y.to_owned()))
-        .collect();
-    let archiving = tokio::task::spawn_blocking(move || {
-        // have to use std here, cause TarBuilder is not async
-        let file = std::fs::File::create(&full_snapshot_path_clone)?;
-        let mut builder = TarBuilder::new(file);
-        for (collection_name, snapshot_details) in created_snapshots_clone {
-            let snapshot_path = snapshot_dir
-                .join(collection_name)
-                .join(&snapshot_details.name);
-            builder.append_path_with_name(&snapshot_path, &snapshot_details.name)?;
-            std::fs::remove_file(snapshot_path)?;
+        let config_path = snapshot_dir.join(format!("config-{current_time}.json"));
+
+        {
+            let snapshot_config = SnapshotConfig {
+                collections_mapping: collection_name_to_snapshot_path,
+                collections_aliases: alias_mapping,
+            };
+            let mut config_file = tokio::fs::File::create(&config_path).await.unwrap();
+            config_file
+                .write_all(
+                    serde_json::to_string_pretty(&snapshot_config)
+                        .unwrap()
+                        .as_bytes(),
+                )
+                .await
+                .unwrap();
         }
-        builder.append_path_with_name(&config_path_clone, "config.json")?;
 
-        builder.finish()?;
-        Ok::<(), StorageError>(())
-    });
+        let full_snapshot_path = snapshot_dir.join(&snapshot_name);
+
+        let config_path_clone = config_path.clone();
+        let full_snapshot_path_clone = full_snapshot_path.clone();
+        let created_snapshots_clone: Vec<_> = created_snapshots
+            .iter()
+            .map(|(x, y)| (x.to_string(), y.to_owned()))
+            .collect();
+        let archiving = tokio::task::spawn_blocking(move || {
+            // have to use std here, cause TarBuilder is not async
+            let file = std::fs::File::create(&full_snapshot_path_clone)?;
+            let mut builder = TarBuilder::new(file);
+            for (collection_name, snapshot_details) in created_snapshots_clone {
+                let snapshot_path = snapshot_dir
+                    .join(collection_name)
+                    .join(&snapshot_details.name);
+                builder.append_path_with_name(&snapshot_path, &snapshot_details.name)?;
+                std::fs::remove_file(snapshot_path)?;
+            }
+            builder.append_path_with_name(&config_path_clone, "config.json")?;
+
+            builder.finish()?;
+            Ok::<(), StorageError>(())
+        });
 
-    archiving.await??;
+        archiving.await.unwrap().unwrap();
```

Comment:
why the `unwrap` here and below in this file?

Reply:
I tried to use `?` firstly, but it will be error.

Reply:
`.unwrap().unwrap()` will kill the whole service